src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_LENGTH = 1 << 4;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_LENGTH];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndexKey(key);
+        Node<K, V> current = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<K, V>(key, value, null);
+            size++;
+            return;
+        }
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            if (current.next == null) {
+                current.next = new Node<>(key, value,null);
+                size++;
+                return;
+            }
+            current = current.next;
+        }
```

Comment:
Refactor it, first if on lines 20 - 24 useless, no need to keep second if inside the while, just move lines 31 - 32 after loop